pkg/ibmcsidriver/server_test.go

```diff
@@ -89,3 +106,72 @@ func TestSetup(t *testing.T) {
 func TestLogGRPC(t *testing.T) {
 	t.Logf("TODO:~ TestLogGRPC")
 }
+
+func TestSetupSidecar(t *testing.T) {
+	tests := []struct {
+		name             string
+		groupID          string
+		fileOps          FileOps
+		expectedErr      bool
+		expectedChownErr error
+		expectedChmodErr error
+	}{
+		{
+			name:             "Successful setupSidecar",
+			groupID:          "123",
+			fileOps:          &mockFileOps{},
+			expectedErr:      false,
+			expectedChownErr: nil,
+			expectedChmodErr: nil,
+		},
+		{
+			name:    "Chown failure",
+			groupID: "123",
+			fileOps: &mockFileOps{
+				chownErr: errors.New("Chown failed"),
+			},
+			expectedErr:      true,
+			expectedChownErr: errors.New("Chown failed"),
+			expectedChmodErr: nil,
+		},
+		{
+			name:    "Chmod failure",
+			groupID: "123",
+			fileOps: &mockFileOps{
+				chmodErr: errors.New("Chmod failed"),
+			},
+			expectedErr:      true,
+			expectedChownErr: nil,
+			expectedChmodErr: errors.New("Chmod failed"),
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			// Set environment variable
+			os.Setenv("SIDECAR_GROUP_ID", tt.groupID)
+
+			// Call setupSidecar
+			err := setupSidecar("someAddr", tt.fileOps)
+
+			// Check the expected error
+			if (err != nil) != tt.expectedErr {
```

Comment:
Please refer other unit tests while validating output
also consider tests when SIDECAR_GROUP_ID env is not set

Reply:
Done